### PR TITLE
Add logging of snapshot information in ResourceMonitorService

### PR DESCRIPTION
**Pull request by evgenyfedorov2**
Add logging of `Snapshot` information in `ResourceMonitorService`
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/extensions/pull/5362).

The whole logging feature will be mentioned in the docs, see https://github.com/dotnet/docs/pull/42300

### CODE REVIEW

File: src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Log.cs [hunk 1]
```diff
@@ -16,4 +16,14 @@ internal static partial class Log
 
     [LoggerMessage(2, LogLevel.Error, "Publisher `{Publisher}` was unable to publish utilization statistics.")]
     public static partial void HandlePublishUtilizationException(ILogger logger, Exception e, string publisher);
+
+    [LoggerMessage(3, LogLevel.Debug,
+    "Snapshot received: TotalTimeSinceStart={totalTimeSinceStart}, KernelTimeSinceStart={kernelTimeSinceStart}, "
+    + "UserTimeSinceStart={userTimeSinceStart}, MemoryUsageInBytes={memoryUsageInBytes}.")]
```
Comment (RussKie) [inline src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Log.cs:22]: Don't break the string, keep it as one-liner.

**Review by RussKie — COMMENTED** [review]
Does this new functionality need to be documented?

File: src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Log.cs (hunk 1 above)
Comment (evgenyfedorov2) [reply src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Log.cs]: Done

**Comment by evgenyfedorov2** [comment]
> Does this new functionality need to be documented?

I don't think so, log statements is not something we usually document. But we can mention it in the docs anyway

**Comment by RussKie** [comment]
> ...we can mention it in the docs anyway

Please link the doc update PR  here.

**Review by RussKie — APPROVED** [review]

**Review by iliar-turdushev — APPROVED** [review]

**Review by mobratil — APPROVED** [review]

**Comment by evgenyfedorov2** [comment]
> > ...we can mention it in the docs anyway
> 
> Please link the doc update PR here.

updated PR description

**Pull request merged**